packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -237,6 +237,7 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     if (!wasBatchTruncated && !this._shouldSubmitBatch(batchSizeInBytes)) {
       return
     }
+    this.bsMetrics.numTxPerBatch.observe(endBlock - startBlock)
```

Comment:
Shouldn't this only be getting measured after the batch is successfully appended?

Reply:
that was my original plan, but when we get the transaction receipt we no longer have access to the `endBlock` and `startBlock` variables. 

passing it through seems heavy for observing this value, and so did using the abi to parse the logs from the receipt. adding this here felt like a decent tradeoff since at this point logically we _should_ be submitting the batch. let me know if you think differently though!

Reply:
Given that we have seen many errors related to submission of batches, I worry this will give us false data. 

Couldn't this be on line 265?

Reply:
^that function may be called multiple times to retry for gas and thus also push the metric multiple times; i'll look into retrieving this value from the logs or elsewhere

Reply:
We could check for the receipt, have an in-memory check to indicate it only runs upon 100% inclusion, etc.

Also, you just made me think that `batchSubmissionRetriesCount` may be a useful metric as well. Ideally we're getting transactions included immediately.

Reply:
i played around with the logs and there isn't a light weight to parse, so ruling that out for now. i think we could potentially filter for `numTxPerBatch` that are observed alongside a submission timestamp metric, so we know which ones actually make it through?

i'll think about the retries count metrics - might be easier to observe when it attempts to submit and when it actually does, which we can deduce from this metric vs the actual submission timestamp 👀